server/src/controllers/ResourcesController.js

```diff
@@ -0,0 +1,170 @@
+const { Resource, Topic } = require("../models/Resources");
+
+module.exports = {
+  /**
+   * @route GET api/topic
+   * @description Retrieve all resources grouped by topics
+   * @param {Object} req The request
+   * @param {Object} res The response
+   */
+  getAllTopics: async (req, res) => {
+    try {
+      const topicsAndResources = await Topic.find({}).populate("resources");
+      res.status(200).json(topicsAndResources);
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route GET api/resource/bytopic/:id
+   * @description Retrieve all resources of a topic by its _id
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {String} req.params.id The _id of the topic
+   */
+  getResourcesByTopicId: async (req, res) => {
+    try {
+      const topics = await Topic.findById(req.params.id).populate("resources");
+      res.status(200).json(topics);
+    } catch (error) {
+      res.status(500).json({ error });
+    }
+  },
+  /**
+   * @route GET api/topic/:id
+   * @description Get a particular topic (and its resources) by its _id
+   * @param {Object} req The request
+   * @param {Object} res The response
+   * @param {String} req.params.id The _id of the topic
+   */
+  getTopicById: async (req, res) => {
+    try {
+      const topic = await Topic.findById(req.params.id).populate("resources");
```

Comment:
This behaviour mimics `getResourcesByTopicId`. If the intention is the same, we should remove one of these commands. Please let us know what you intended with `getTopicById`.

Reply:
Oh right, makes sense. I missed that. I'll remove `getResourcesByTopicId` and its endpoint